docs/Migration-Guide-V3.md

```diff
@@ -0,0 +1,100 @@
+# V3 Migration Guide
+
+This guide is aimed to help migrating from Fastify v2 to v3.
+
+Before beginning please ensure that any deprecation warnings from v2 are fixed as we have removed all deprecated features and they will no longer work after upgrading. (#1750)
+
+## Breaking changes
+
+### Changed middleware support (#2014)
+
+From Fastify v3, middleware support does not come out of the box with the framework itself.
+
+If you use Express middleware in your application, please install and register the [`fastify-express`](https://github.com/fastify/fastify-express) or [`middie`](https://github.com/fastify/middie) plugin before doing so.
+
+**v2:**
+
+```js
+fastify.use(require("cors")());
+```
+
+**v3:**
+
+```js
+await fastify.register(require("fastify-express"));
+fastify.use(require("cors")());
+```
+
+### Changed schema substitution (#2023)
+
+We have dropped support for non-standard `replace-way` shared schema substitution and replaced it with standard compliant `$ref-way` references.
```

Comment:
Please add a link to the documentation of the "replace-way" behavior. Also, `$ref-way` -> `$ref`. The standard is JSON Schema.